src/Compilers/Core/Portable/SourceGeneration/RunResults.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Text;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents the results of running a generation pass over a set of <see cref="ISourceGenerator"/>s.
+    /// </summary>
+    public readonly struct GeneratorDriverRunResult
+    {
+        internal GeneratorDriverRunResult(ImmutableArray<GeneratorRunResult> results, ImmutableArray<SyntaxTree> trees, ImmutableArray<Diagnostic> diagnostics)
+        {
+            this.Results = results;
+            this.SyntaxTrees = trees;
+            this.Diagnostics = diagnostics;
+        }
+
+        /// <summary>
+        /// The individual result of each <see cref="ISourceGenerator"/> that was run in this generator pass, one per generator.
+        /// </summary>
+        public ImmutableArray<GeneratorRunResult> Results { get; }
+
+        /// <summary>
+        /// The <see cref="Diagnostic"/>s produced by all generators run during this generation pass.
+        /// </summary>
+        /// <remarks>
+        /// This is equivalent to the union of all <see cref="GeneratorRunResult.Diagnostics"/> in <see cref="Results"/>.
+        /// </remarks>
+        public ImmutableArray<Diagnostic> Diagnostics { get; }
+
+        /// <summary>
+        /// The <see cref="SyntaxTree"/>s produced during this generation pass by parsing each <see cref="SourceText"/> added by each generator.
+        /// </summary>
+        /// <remarks>
+        /// This is equivalent to the union of all <see cref="GeneratedSourceResult.SyntaxTree"/>s in each <see cref="GeneratorRunResult.AddedSources"/> in each <see cref="Results"/>
+        /// </remarks>
+        public ImmutableArray<SyntaxTree> SyntaxTrees { get; }
```

Comment:
>SyntaxTrees [](start = 42, length = 11)

I feel like `GeneratedTrees` would be a better name.

Reply:
Not addressed?

---
In reply to: [466075083](https://github.com/dotnet/roslyn/pull/46296#discussion_r466075083) [](ancestors = 466075083)

Reply:
I don't know if we're going to actually keep this or not. Thoughts on [https://github.com/dotnet/roslyn/pull/46296#discussion_r466089178](https://github.com/dotnet/roslyn/pull/46296#discussion_r466089178)

---
In reply to: [466573790](https://github.com/dotnet/roslyn/pull/46296#discussion_r466573790) [](ancestors = 466573790,466075083)